Assets/MixedRealityToolkit/Inspectors/MixedRealityWireframeShaderGUI.cs

```diff
@@ -0,0 +1,210 @@
+
+            cullMode = FindProperty("_CullMode", props);
+            renderQueueOverride = FindProperty("_RenderQueueOverride", props);
+
+            baseColor = FindProperty("_BaseColor", props);
+            wireColor = FindProperty("_WireColor", props);
+            wireThickness = FindProperty("_WireThickness", props);
+        }
+
+        public override void AssignNewShaderToMaterial(Material material, Shader oldShader, Shader newShader)
+        {
+            float? cullMode = GetFloatProperty(material, "_Cull");
+
+            base.AssignNewShaderToMaterial(material, oldShader, newShader);
+   
+            SetFloatProperty(material, null, "_CullMode", cullMode);
+
+            // Setup the rendering mode based on the old shader.
+            if (oldShader == null || !oldShader.name.Contains("Legacy Shaders/"))
+            {
+                SetupMaterialWithRenderingMode(material, (RenderingMode)material.GetFloat(BaseStyles.renderingModeName), CustomRenderingMode.Opaque, -1);
+            }
+            else
+            {
+                RenderingMode mode = RenderingMode.Opaque;
+
+                if (oldShader.name.Contains("/Transparent/Cutout/"))
+                {
+                    mode = RenderingMode.TransparentCutout;
+                }
+                else if (oldShader.name.Contains("/Transparent/"))
+                {
+                    mode = RenderingMode.Transparent;
+                }
+
+                material.SetFloat(BaseStyles.renderingModeName, (float)mode);
+
+                MaterialChanged(material);
+            }
+        }
+
+        protected void Initialize(Material material)
+        {
+            if (!initialized)
+            {
+                MaterialChanged(material);
+                initialized = true;
+            }
+        }
+
+        protected void MaterialChanged(Material material)
+        {
+            SetupMaterialWithRenderingMode(material, 
+                (RenderingMode)renderingMode.floatValue, 
+                (CustomRenderingMode)customRenderingMode.floatValue, 
+                (int)renderQueueOverride.floatValue);
+        }
+
+        // TODO: Merge this out? with StandardShaderGUI
+        protected void RenderingModeOptions(MaterialEditor materialEditor)
+        {
+            EditorGUI.BeginChangeCheck();
+
+            EditorGUI.showMixedValue = renderingMode.hasMixedValue;
+            RenderingMode mode = (RenderingMode)renderingMode.floatValue;
+            EditorGUI.BeginChangeCheck();
+            mode = (RenderingMode)EditorGUILayout.Popup(renderingMode.displayName, (int)mode, BaseStyles.renderingModeNames);
+
+            if (EditorGUI.EndChangeCheck())
+            {
+                materialEditor.RegisterPropertyChangeUndo(renderingMode.displayName);
+                renderingMode.floatValue = (float)mode;
+            }
+
+            EditorGUI.showMixedValue = false;
+
+            if (EditorGUI.EndChangeCheck())
+            {
```

Comment:
More general question but, is is there anything weird with having unapired EndChangeCheck()s (i.e. one begin, multiple ends?)

Reply:
I didn't even catch this. It is original code. But I don't like the double end with one beginning either, even if it works. I'll fix